Hw04

SergeevED opened the pull request.

SergeevED committed:
hw03

hw03

SergeevED committed:
hw03

hw03

SergeevED committed:
hw03

fixed

SergeevED committed:
hw03

fixed

SergeevED committed:
hw03

fixed

SergeevED committed:
hw03

small fix RemoveAt()

SergeevED committed:
hw03

small bagfix

SergeevED committed:
hw03

fixed

SergeevED committed:
hw03

fixed

SergeevED committed:
hw04

hw04

SergeevED committed:
hw04

anlun commented on 2st semester/hw04/Graph_with_unit_tests.fs:
```diff
@@ -0,0 +1,349 @@
+  let vertices = [||]
+  let adjMatrix = Array2D.create 0 0 true
+  let mGraph = (new MatrixGraph<int>(vertices, adjMatrix)) :> IGraph<int>
+  Assert.AreEqual([], GetSuccessors mGraph 0)
```
Тесты, которые нужно объединить через TestCase.

anlun commented on 2st semester/hw04/LAN_with_unit_tests.fs:
```diff
@@ -0,0 +1,200 @@
+              { number = 3; os = "3"; infectedChance = 0; isInfected = false} 
+              { number = 4; os = "4"; infectedChance = 0; isInfected = true} 
+              { number = 5; os = "5"; infectedChance = 0; isInfected = false} 
+           |]
```
Тесты стоит объединить через TestCase. И стоит завести test-class, в котором массивы l будут
полями.

anlun commented on 2st semester/hw04/Polymorphous_lists_with_unit_tests.fs:
```diff
@@ -0,0 +1,544 @@
+  Assert.AreEqual(true, b1)
+  Assert.AreEqual(true, b2)
+  Assert.AreEqual(true ,b3)
+  Assert.AreEqual([|'b';'c';'e'|] , listToArr mList)
```
Несколько AreEqual -- подозрение на несколько на несколько тестов.

anlun commented on 2st semester/hw04/Stack_calculator_with_unit_tests.fs:
```diff
@@ -0,0 +1,290 @@
+  Assert.AreEqual(None, stackCalc str2 [])
+  Assert.AreEqual(None, stackCalc str3 values)
+  Assert.AreEqual(None, stackCalc str4 [])
+  Assert.AreEqual(None, stackCalc str5 [])
```
Нужно упростить с помощью TestCase.

anlun commented on 2st semester/hw04/Stack_calculator_with_unit_tests.fs:
```diff
@@ -0,0 +1,290 @@
+        else
+          operatorWasScanned <- true
+          while (not (tempStack.Size() = 0)) &&
+          (isLeftAssoc (x.Chars(0)) && (precedence (x.Chars(0)) <= precedence (Option.get (tempStack.Top() )))
```
Слишком сложное условие. Нужно сделать функцией и в ней ввести несколько переменных для подвыражений.

SergeevED committed:
hw04. 30.03.2015

Структурированы юнит тесты полиморфного списка.
Структурированы юнит тесты стекового калькулятора.
Небольшие изменения структуры стекового калькулятора:
- Stack.Pop() возвращает 'A вместо Some 'A
- логика формирования постфиксной записи из инфиксной вынесена в
отдельную функцию
firstOperatorAfterSecond(op1, op2)

SergeevED committed:
hw04 01.04.2015

Исправлена ошибка в функции calculate
Добавлен юнит тест для пустой строки
Правка форматирования кода
